examples/energy_mng/ui/widgets/scroll_view.slint

```diff
@@ -0,0 +1,118 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+import { Theme } from "../theme.slint";
+import { StateLayer } from "../components/state_layer.slint";
+
+component ScrollBar inherits Rectangle {
+    in-out property <bool> horizontal;
+    in-out property<length> maximum;
+    in-out property<length> page-size;
+    // this is always negative and bigger than  -maximum
+    in-out property<length> value;
+
+    background: Theme.palette.pure-black;
+    border-width: 1px;
+
+    i-handle := Rectangle {
+        width: !root.horizontal ? parent.width : root.maximum <= 0phx ? 0phx : parent.width * (root.page-size / (root.maximum + root.page-size));
+        height: root.horizontal ? parent.height : root.maximum <= 0phx ? 0phx : parent.height * (root.page-size / (root.maximum + root.page-size));
+
+        border-radius: 3px;
+        background: Theme.palette.slint-blue-400;
+        x: !root.horizontal ? 0phx : (root.width - i-handle.width) * (-root.value / root.maximum);
+        y: root.horizontal ? 0phx : (root.height - i-handle.height) * (-root.value / root.maximum);
+    }
+
+    i-touch-area := StateLayer {
+        private property <length> pressed-value;
+
+        width: parent.width;
+        height: parent.height;
+       
+        pointer-event(event) => {
+            if (event.button == PointerEventButton.left && event.kind == PointerEventKind.down) {
+                self.pressed-value = -root.value;
+            }
+        }
+        moved => {
+            if (self.enabled && self.pressed) {
+                root.value = -max(0px, min(root.maximum, self.pressed-value + (
+                    root.horizontal ? (i-touch-area.mouse-x - i-touch-area.pressed-x) * (root.maximum / (root.width - i-handle.width))
+                               : (i-touch-area.mouse-y - i-touch-area.pressed-y) * (root.maximum / (root.height - i-handle.height))
+                )));
+            }
+        }
+    }
+}
+
+export component ScrollView {
+    in-out property <length> viewport-width <=> i-flickable.viewport-width;
+    in-out property <length> viewport-height <=> i-flickable.viewport-height;
+    in-out property <length> viewport-x <=> i-flickable.viewport-x;
+    in-out property <length> viewport-y <=> i-flickable.viewport-y;
+    out property <length> visible-width <=>  i-flickable.width;
+    out property <length> visible-height <=>  i-flickable.height;
+    in property <bool> enabled: true;
+    // FIXME: remove. This property is currently set by the ListView and is used by the native style to draw the scrollbar differently when it has focus
+    in-out property <bool> has-focus;
+    
+    min-height: 50px;
+    min-width: 50px;
+    horizontal-stretch: 1;
+    vertical-stretch: 1;
+
+    i-flickable := Flickable {
+        x: 2px;
+        y: 2px;
+        interactive: false;
```

Comment:
On embedded, we want this to be interactive

Reply:
done